dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -105,30 +105,41 @@ def refute_estimate(self):
         self.logger.info("Refutation over {} simulated datasets".format(self._num_simulations) )
         self.logger.info("The transformation passed: {}".format(self._transformations) )
 
-        data_chunks = self.preprocess_data_by_treatment()
+        groups = self.preprocess_data_by_treatment()
         estimates = []
-        for chunk in data_chunks:
-
-            X_input = chunk[self._chosen_variables]
-            new_outcome = chunk['y']
-            X = self._data[self._chosen_variables]
+        for key_train, _ in groups:

```

Comment:
Here's a simpler and more readable way to achieve this, without having two nested loops operating on groups, which is a complex object and we don't how it will behave in nested loops. It uses the fact that get_group can take in multiple keys.
1. First, make an array of all keys : keys=[k for k,_ in groups]
2. for k in keys:  
             train_Df = groups.get_group(k); 
             validate_df = groups.get_group( set(keys).difference(set(k)) )
             # Now you can write all other code.

Reply:
That is a clean way to do it. Nice, I'll incorporate it into the code.